diff --git a/thirdparty/hev-socks5-tunnel-2.6.7/src/misc/hev-exec.c b/thirdparty/hev-socks5-tunnel-2.6.7/src/misc/hev-exec.c
index 32158e9..1a76d59 100644
--- a/thirdparty/hev-socks5-tunnel-2.6.7/src/misc/hev-exec.c
+++ b/thirdparty/hev-socks5-tunnel-2.6.7/src/misc/hev-exec.c
@@ -16,15 +16,18 @@
 
 #include "hev-exec.h"
 
+#if 0
 static void
 signal_handler (int signum)
 {
     waitpid (-1, NULL, WNOHANG);
 }
+#endif
 
 void
 hev_exec_run (const char *script_path, const char *tun_name, int wait)
 {
+#if 0
     pid_t pid;
 
     signal (SIGCHLD, signal_handler);
@@ -40,7 +43,9 @@ hev_exec_run (const char *script_path, const char *tun_name, int wait)
     }
 
     execl (script_path, script_path, tun_name, NULL);
-
-    LOG_E ("exec %s %s", script_path, tun_name);
-    exit (-1);
+#else
+    (void)wait;
+    LOG_E("exec %s %s", script_path, tun_name);
+    exit(-1);
+#endif
 }
diff --git a/thirdparty/hev-socks5-tunnel-2.6.7/src/misc/hev-utils.c b/thirdparty/hev-socks5-tunnel-2.6.7/src/misc/hev-utils.c
index 7d6c50a..e65e466 100644
--- a/thirdparty/hev-socks5-tunnel-2.6.7/src/misc/hev-utils.c
+++ b/thirdparty/hev-socks5-tunnel-2.6.7/src/misc/hev-utils.c
@@ -20,6 +20,7 @@
 void
 run_as_daemon (const char *pid_file)
 {
+#if 0
     FILE *fp;
 
     fp = fopen (pid_file, "w+");
@@ -37,6 +38,9 @@ run_as_daemon (const char *pid_file)
 
     fprintf (fp, "%u\n", getpid ());
     fclose (fp);
+#else
+    (void)pid_file;
+#endif
 }
 
 int
diff --git a/thirdparty/hev-socks5-tunnel-2.6.7/third-part/lwip/include/ports/unix/lwipopts.h b/thirdparty/hev-socks5-tunnel-2.6.7/third-part/lwip/include/ports/unix/lwipopts.h
index 3aa62ac..9018fd4 100644
--- a/thirdparty/hev-socks5-tunnel-2.6.7/third-part/lwip/include/ports/unix/lwipopts.h
+++ b/thirdparty/hev-socks5-tunnel-2.6.7/third-part/lwip/include/ports/unix/lwipopts.h
@@ -51,6 +51,8 @@
    -----------------------------------------------
 */
 
+#define LWIP_HAVE_SLIPIF 0
+
 /**
  * SYS_LIGHTWEIGHT_PROT==1: if you want inter-task protection for certain
  * critical regions during buffer allocation, deallocation and memory
